### PR TITLE
Check for Resource Already Exist Error before retry

### PR DESCRIPTION
**Pull request by uzaxirr**
Fixes: #234

### CODE REVIEW

File: civo/instances/resource_instance.go [hunk 1]
```diff
@@ -278,19 +279,30 @@ func resourceInstanceCreate(ctx context.Context, d *schema.ResourceData, m inter
...
-			return err
-		}
-		d.SetId(instance.ID)
-		return nil
-	}, 10*time.Second, 2*time.Minute)
 
+	instance, err := apiClient.CreateInstance(config)
 	if err != nil {
-		return diag.Errorf("[ERR] failed to create instance after multiple attempts: %s", err)
+		if errors.Is(err, civogo.DatabaseInstanceDuplicateNameError) {
```
Comment (vishalanarase) [inline civo/instances/resource_instance.go:285]: Is this the only error we want to verify? There are multiple errors are present, What will happen to those?

File: civo/network/resource_network.go [hunk 2]
```diff
@@ -129,21 +130,32 @@ func resourceNetworkCreate(ctx context.Context, d *schema.ResourceData, m interf
...
-		log.Printf("[INFO] Attempting to create the network %s", d.Get("label").(string))
-		network, err := apiClient.CreateNetwork(configs)
-		if err != nil {
-			return err
+	log.Printf("[INFO] Attempting to create the network %s", d.Get("label").(string))
+	network, err := apiClient.CreateNetwork(configs)
+	if err != nil {
+
+		// Check for the resource already exists error
+		if errors.Is(err, civogo.DatabaseNetworkExistsError) {
```
Comment (vishalanarase) [inline civo/network/resource_network.go:138]: Same comment as above

**Comment by uzaxirr** [comment]
@vishalanarase there are a ton of errors in the SDK not feasible to check for all of them. Duplicate name is the most common one hence checking for that only to improve the experience. for other errors tf will error out after all the retry. But for Duplicate Resource it will error out immediately without retry.

**Review by fernando-villalba — APPROVED** [review]
This is good, but there may be other resources aside from network and instances, may be worth checking kubernetes, and others as well.

**Pull request merged**